catalog/format/iceberg-bench/build.gradle.kts

```diff
@@ -17,7 +17,7 @@
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
 plugins {
-  id("nessie-conventions-client")
+  id("nessie-conventions-server")
```

Comment:
the change looks a bit odd as in other places we change from `nessie-conventions-server` to `nessie-conventions-client` :thinking: 

are we saying both client and server components for iceberg now require j11 ?
or does this fix a previous misconfiguration of using client for a server component?

Reply:
Iceberg requires Java >=11 now

Reply:
yes but that explains why the convention themselves would change, not why iceberg-bench would switch from client to server